### ISSUE TITLE
[VueJS] amplify-connect passing authMode

### ISSUE THREAD

**Issue by DanielMuller** [aws-amplify/amplify-js#4713]
** Which Category is your question related to? **
API
** What AWS Services are you utilizing? **
Appsync, Cognito
** Provide additional details e.g. code snippets **
I am defining a model with multi-auth for public read (Cognito_identity_pool and IAM). Cognito_identity_pool is set as the default authorization
```
type Album @model
@auth(rules:[
  { allow: owner },
  { allow: public, provider: iam, operations: [read] }
])
{
  id: ID!
  name: String!
}
```
I want to use the `amplify-connect` component to execute my query.
This work fine when the user is authenticated, but for unAuth users, this fails. Normal, since I can't pass `authMode` to the component.

Using directly `this.$Amplify.API.graphql({query, variables, authMode: 'AWS_IAM'})` works.

My questions: how can I use `amplify-connect` with authMode?
- My first thought, was to try to do a PR and add `authMode` as a new props
- Is there another way that is recommended, like setting a parameter?
- I tried to call `$Amplify.API.configure({ 'aws_appsync_authenticationType': 'AWS_IAM' })` before creating the query when no user is loggued in. This works, but seems very dirty.

Thanks,
Daniel

**Comment by amhinson** [comment]
@DanielMuller You are correct in that `amplify-connect` currently doesn't have the ability to change the `authMode` of the request, which means that it will always fallback to the default auth mode for the API, which is Cognito User Pools. The workaround you provided with calling `API.graphql()` directly would be the recommended path right now to achieve the functionality you're looking for.

I'm labeling this as a feature request for now so we can take it into account for any further updates regarding UI components & `amplify-connect` 👍
